src/oom/test-oomd-util.c

```diff
@@ -0,0 +1,359 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <unistd.h>
+
+#include "alloc-util.h"
+#include "cgroup-setup.h"
+#include "cgroup-util.h"
+#include "fileio.h"
+#include "fs-util.h"
+#include "oomd-util.h"
+#include "parse-util.h"
+#include "string-util.h"
+#include "strv.h"
+#include "tests.h"
+
+static int fork_and_sleep(unsigned sleep_min) {
+        usec_t n, timeout, ts;
+
+        pid_t pid = fork();
+        assert_se(pid >= 0);
+
+        if (pid == 0) {
+                timeout = sleep_min * USEC_PER_MINUTE;
+                ts = now(CLOCK_MONOTONIC);
+                while (true) {
+                        n = now(CLOCK_MONOTONIC);
+                        if (ts + timeout < n) {
+                                log_error("Child timed out waiting to be killed");
+                                abort();
+                        }
+                        sleep(1);
+                }
+        }
+
+        return pid;
+}
+
+static void systemd_cgroup_clean(char **p) {
+        if (!p)
+                return;
+
+        cg_rmdir(SYSTEMD_CGROUP_CONTROLLER, *p);
+        free(*p);
+}
+
+static void test_oomd_cgroup_kill(void) {
+        _cleanup_free_ char *vi = NULL, *vd = NULL, *full_path = NULL;
+        _cleanup_(systemd_cgroup_clean) char *path = NULL;
+        char template[] = "/oomdkilltestXXXXXX";
+        int pid[2];
+        char *p;
+
+        if (geteuid() != 0)
+                return (void) log_tests_skipped("not root");
+
+        if (cg_all_unified() <= 0)
+                return (void) log_tests_skipped("cgroups are not running in unified mode");
+
+        assert_se(cg_get_path(SYSTEMD_CGROUP_CONTROLLER, template, NULL, &full_path) == 0);
```

Comment:
@poettering added a helper to allocate a cgroup for tests in b7b7ace4eae6d64e520d037f02f1177eac063eb0. It'd be nice to use it here too.